### PR TITLE
Add un-clickable backdrop to Modal

### PR DESCRIPTION
**Pull request by ChirashiSushi**
### Please review these reminders and either check the box or explain why not:

- [x] Read and followed the [contributing guidelines](https://eds.eks.dev.ethos-int.com/#/Guidelines?id=section-contribute)?
- [x] Component is exported from [index.js](https://github.com/getethos/ethos-design-system/blob/master/src/components/index.js)?
- [x] Type is exported from [index.d.ts](https://github.com/getethos/ethos-design-system/blob/master/src/components/index.d.ts) so Typescript users can consume it? Added a test and ran `yarn test:types`?
- [x] Bumped the yarn version?

## **Screenshot**

https://user-images.githubusercontent.com/84357906/183788908-f17de6df-2595-4b90-aabb-b531acf1eed8.mov

### CODE REVIEW

File: src/components/Modal/Modal.js [hunk 1]
```diff
@@ -72,6 +72,15 @@ export const Modal = ({
...
 }) => {
   const classes = isOpen ? styles.ModalWrapperActive : styles.ModalWrapper
+  const [backDropClass, setBackDropClass] = useState(styles.ModalBackdrop)
+
+  useEffect(() => {
+    setTimeout(() => {
+      setBackDropClass(
+        isOpen ? styles.ModalBackdropActive : styles.ModalBackdrop
+      )
+    }, 1000)
```
Comment (coleEthos) [inline src/components/Modal/Modal.js:82]: Why are we waiting a full second here?

File: src/components/Modal/Modal.js [hunk 2]
```diff
@@ -72,6 +72,15 @@ export const Modal = ({
...
+  })
```
Comment (coleEthos) [inline src/components/Modal/Modal.js:83]: Add empty dependency array?

**Review by coleEthos — COMMENTED** [review]
This seems like an okay approach. Couple minor questions above. Going to pull the branch and do a little bit of local QA before approving.

**Review by coleEthos — CHANGES_REQUESTED** [review]
I think we need to lower that timeout value from 1 second to be much less or find a way that doesn't rely on a timeout. If a user closes the modal and tries to click something right away, the backdrop blocks them from clicking. I was able to reproduce this locally and verify by increasing to 5000ms in the timeout. 1000 is less noticeable but still could be an issue.

File: src/components/Variables.scss [hunk 3]
```diff
@@ -5,5 +5,6 @@
 $z-index-navbar: 1;
 $z-index-middle: 2;
 $z-index-modal: 3;
+$z-index-backdrop: 1000;
```
Comment (WesleyDRobinson) [inline src/components/Variables.scss:8]: 🦖 
⛰️

File: src/components/Modal/Modal.js (hunk 2 above)
Comment (ChirashiSushi) [reply src/components/Modal/Modal.js:83]: Adding empty dependency array here actually makes `useEffect` not working for some reason.

Comment (coleEthos) [reply src/components/Modal/Modal.js:83]: I'm not sure why I suggested that. I'm realizing now this probably needs to run every render not just on the first one.

**Review by coleEthos — APPROVED** [review]
LGTM, thanks for lowering the timeout. We should be sure to double check the components that consumer this when we update CMS & main app to check for any special cases where this might cause issues, but based on testing locally in EDS alone, it seems to work fine!

**Pull request merged**